src/snowflake/snowpark/functions.py

```diff
@@ -2014,6 +2014,47 @@ def div0(
     return builtin("div0", _emit_ast=_emit_ast)(dividend_col, divisor_col)
 
 
+@publicapi
+def divnull(
+    dividend: Union[ColumnOrName, int, float],
+    divisor: Union[ColumnOrName, int, float],
+    _emit_ast: bool = True,
+) -> Column:
+    """Performs division like the division operator (/),
+    but returns NULL when the divisor is 0 (rather then reporting error).
+
+    Example::
+
+        >>> df = session.create_dataframe([1], schema=["a"])
+        >>> df.select(divnull(df["a"], 1).alias("divided_by_one"), divnull(df["a"], 0).alias("divided_by_zero")).collect()
+        [Row(DIVIDED_BY_ONE=Decimal('1.000000'), DIVIDED_BY_ZERO=None)]
+    """
+    dividend_col = (
+        lit(dividend, _emit_ast=False)
+        if isinstance(dividend, (int, float))
+        else _to_col_if_str(dividend, "divnull")
+    )
+    divisor_col = (
+        lit(divisor, _emit_ast=False)
+        if isinstance(divisor, (int, float))
+        else _to_col_if_str(divisor, "divnull")
+    )
+    return dividend_col / nullifzero(divisor_col, _emit_ast=False)
```

Comment:
The problem with this is that in there are valid cases of divisoin that return 0, for example 0/100, 0/20

and with this approach we wont be able to distinguish those from failed ones. Also this we can parity with spark behavior

Reply:
I just understood the nullifzero on the divisor that is great as divisions by zero will then be null